### PR TITLE
[macOS, windows] Adding Swig

### PR DESCRIPTION
**Pull request by kassannk**
**Description**

Adding Swig 4.0.2 for Windows / MacOS

**Check list**

- [x] Windows Server 2016
- [x] Windows Server 2019
- [x] MacOS 10.15
- [x] MacOS 11

Fixes issue:
#3342

### CODE REVIEW

File: images/macos/software-report/SoftwareReport.Common.psm1 [hunk 1]
```diff
@@ -487,6 +487,11 @@ function Get-PowershellVersion {
     return $powershellVersion
 }
 
+function Get-SwigVersion {
+    $swigVersion = Run-Command "swig --version" -SuppressStderr
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Common.psm1:491]: What is the output of `swig --version`? Does it return only version or any other info?

File: images/macos/software-report/SoftwareReport.Generator.ps1 [hunk 2]
```diff
@@ -159,7 +159,8 @@ $toolsList = @(
     (Get-AWSSAMCLIVersion),
     (Get-AWSSessionManagerCLIVersion),
     (Get-AliyunCLIVersion),
-    (Get-XcodeCommandLineToolsVersion)
+    (Get-XcodeCommandLineToolsVersion),
+    (Get-SwigVersion)
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Generator.ps1:163]: Could you please add test for swig on macOS too?

File: images/win/scripts/Installers/Install-swig.ps1 [hunk 3]
```diff
@@ -0,0 +1,8 @@
+################################################################################
+##  File:  Install-swig.ps1
+##  Desc:  Install swig
+################################################################################
+
+Choco-Install -PackageName swig.install
```
Comment (maxim-lobanov) [inline images/win/scripts/Installers/Install-swig.ps1:6]: I can't find choco package with name `swig.install`. I see only `swig`: https://community.chocolatey.org/packages/swig

File: images/win/scripts/SoftwareReport/SoftwareReport.Tools.psm1 [hunk 4]
```diff
@@ -248,13 +248,18 @@ function Get-VisualCPPComponents {
...
                 Version = $version
-            }
-        }
-    }
-}
-
+ }
+            
 function Get-DacFxVersion {
     cd "C:\Program Files\Microsoft SQL Server\150\DAC\bin\"
```
Comment (maxim-lobanov) [inline images/win/scripts/SoftwareReport/SoftwareReport.Tools.psm1:254]: Looks like brackets are broken here. Could you please double check?

File: images/win/windows2016.json [hunk 5]
```diff
@@ -254,6 +254,7 @@
                 "{{ template_dir }}/scripts/Installers/Install-IEWebDriver.ps1",
                 "{{ template_dir }}/scripts/Installers/Install-Apache.ps1",
                 "{{ template_dir }}/scripts/Installers/Install-Nginx.ps1"
+                "{{ template_dir }}/scripts/Installers/Install-Install-swig.ps1"
```
Comment (maxim-lobanov) [inline images/win/windows2016.json:257]: typo: double `Install` in file name

File: images/macos/software-report/SoftwareReport.Common.psm1 (hunk 1 above)
Comment (miketimofeev) [reply images/macos/software-report/SoftwareReport.Common.psm1]: @kassannk have you tested it? There is no `--version` flag, there should be only one dash `-version`

File: images/win/scripts/Tests/Tools.Tests.ps1 [hunk 6]
```diff
@@ -215,6 +215,12 @@ Describe "Stack" {
     }
 }
 
+Describe "swig" {
```
Comment (miketimofeev) [inline images/win/scripts/Tests/Tools.Tests.ps1:218]:
```suggestion
Describe "Swig" {
```

File: images/win/scripts/Tests/Tools.Tests.ps1 [hunk 7]
```diff
@@ -215,6 +215,12 @@ Describe "Stack" {
...
+    It "swig" {
```
Comment (miketimofeev) [inline images/win/scripts/Tests/Tools.Tests.ps1:219]:
```suggestion
    It "Swig" {
```

File: images/win/scripts/SoftwareReport/SoftwareReport.Tools.psm1 [hunk 8]
```diff
@@ -248,13 +248,18 @@ function Get-VisualCPPComponents {
...
     return "DacFx $dacfxversion"
-}
\ No newline at end of file
+}
+
+function Get-SwigVersion {
+    (swig | Out-String) -match "swig (?<version>\d+\.\d+\.?\d*)" | Out-Null
+    $version = $Matches.Version
+    return "swig $version"
+          }
```
Comment (al-cheb) [inline images/win/scripts/SoftwareReport/SoftwareReport.Tools.psm1:263]: Should we remove 264 and 265 line?

File: images/win/scripts/SoftwareReport/SoftwareReport.Tools.psm1 [hunk 9]
```diff
@@ -248,13 +248,18 @@ function Get-VisualCPPComponents {
                 Name = $name
                 Architecture = $arch
                 Version = $version
-            }
-        }
-    }
-}
-
+ }
```
Comment (al-cheb) [inline images/win/scripts/SoftwareReport/SoftwareReport.Tools.psm1:251]: Why do we remove `}` for this function?

File: images/win/windows2016.json [hunk 10]
```diff
@@ -254,6 +254,7 @@
                 "{{ template_dir }}/scripts/Installers/Install-IEWebDriver.ps1",
                 "{{ template_dir }}/scripts/Installers/Install-Apache.ps1",
                 "{{ template_dir }}/scripts/Installers/Install-Nginx.ps1"
+                "{{ template_dir }}/scripts/Installers/Install-swig.ps1"
```
Comment (al-cheb) [inline images/win/windows2016.json:257]: nitpick: Install-Swig.ps1

Comment (kassannk) [reply images/win/windows2016.json:257]: Thanks. File name is Install-swig.ps1, not a Install-Swig.ps1.

Comment (al-cheb) [reply images/win/windows2016.json:257]: To be consistence with other scripts name. I think better to rename Install-swig.ps1 -> Install-Swig.ps1

File: images/macos/software-report/SoftwareReport.Common.psm1 (hunk 1 above)
Comment (kassannk) [reply images/macos/software-report/SoftwareReport.Common.psm1]: > @kassannk have you tested it? There is no `--version` flag, there should be only one dash `-version`

I have already fixed this in the case of windows and mac

File: images/win/windows2019.json [hunk 11]
```diff
@@ -258,6 +258,7 @@
                 "{{ template_dir }}/scripts/Installers/Install-IEWebDriver.ps1",
                 "{{ template_dir }}/scripts/Installers/Install-Apache.ps1",
                 "{{ template_dir }}/scripts/Installers/Install-Nginx.ps1"
+                "{{ template_dir }}/scripts/Installers/Install-Install-Swig.ps1"
```
Comment (AlenaSviridenko) [inline images/win/windows2019.json:261]: typo in the name

File: images/win/scripts/SoftwareReport/SoftwareReport.Tools.psm1 (hunk 8 above)
Comment (AlenaSviridenko) [reply images/win/scripts/SoftwareReport/SoftwareReport.Tools.psm1:266]: Extra spaces before closing bracket

File: images/win/scripts/SoftwareReport/SoftwareReport.Tools.psm1 [hunk 12]
```diff
@@ -248,13 +248,19 @@ function Get-VisualCPPComponents {
                 Name = $name
                 Architecture = $arch
                 Version = $version
-            }
+             }
```
Comment (al-cheb) [inline images/win/scripts/SoftwareReport/SoftwareReport.Tools.psm1:251]: nitpick: remove extra space

File: images/macos/software-report/SoftwareReport.Common.psm1 [hunk 13]
```diff
@@ -487,6 +487,11 @@ function Get-PowershellVersion {
     return $powershellVersion
 }
 
+function Get-SwigVersion {
+    $swigVersion = Run-Command "swig -version"
+    return "Swig $swigVersion"
```
Comment (al-cheb) [inline images/macos/software-report/SoftwareReport.Common.psm1:492]: I think we should correct to parse and display swig version.

**Pull request closed without merging**